src/core/Akka.Persistence.Tests/PersistentActorSpecAsyncAwait.Actors.cs

```diff
@@ -0,0 +1,1337 @@
+                            Context.Become(NewBehavior);
+                        });
+                    });
+                    return true;
+                }
+                return false;
+            }
+
+            protected bool NewBehavior(object message)
+            {
+                if (message is Cmd cmd)
+                {
+                    RunTask(async () =>
+                    {
+                        await Task.Yield();
+                        Persist(new Evt(cmd.Data + "-21"), evt =>
+                        {
+                            UpdateState(evt);
+                            Context.UnbecomeStacked();
+                        });
+                        Persist(new Evt(cmd.Data + "-22"), UpdateStateHandler);
+                    });
+                    return true;
+                }
+                return false;
+            }
+        }
+        internal class ChangeBehaviorInCommandHandlerFirstActor : ExamplePersistentActor
+        {
+            public ChangeBehaviorInCommandHandlerFirstActor(string name) : base(name) { }
+
+            protected override bool ReceiveCommand(object message)
+            {
+                if (CommonBehavior(message)) return true;
+
+                if (message is Cmd cmd)
+                {
+                    RunTask(async () =>
+                    {
+                        await Task.Yield();
+                        Context.Become(NewBehavior);
+                        Persist(new Evt(cmd.Data + "-0"), UpdateStateHandler);
+                    });
+                    return true;
+                }
+                return false;
+            }
+
+            protected bool NewBehavior(object message)
+            {
+                if (message is Cmd cmd)
+                {
+                    RunTask(async () =>
+                    {
+                        await Task.Yield();
+                        Context.UnbecomeStacked();
+                        PersistAll(new[] { new Evt(cmd.Data + "-31"), new Evt(cmd.Data + "-32") }, UpdateStateHandler);
+                        UpdateState(new Evt(cmd.Data + "-30"));
+                    });
+                    return true;
+                }
+                return false;
+            }
+        }
+
+        internal class ChangeBehaviorInCommandHandlerLastActor : ExamplePersistentActor
+        {
+            public ChangeBehaviorInCommandHandlerLastActor(string name) : base(name) { }
+
+            protected override bool ReceiveCommand(object message)
+            {
+                if (CommonBehavior(message)) return true;
+
+                if (message is Cmd cmd)
+                {
+                    RunTask(async () =>
+                    {
+                        await Task.Yield();
+                        Persist(new Evt(cmd.Data + "-0"), UpdateStateHandler);
+                        Context.Become(NewBehavior);
+                    });
+                    return true;
```

Comment:
I don't think it's fully valid way of calling RunTask. As it won't be invoked immediately, the handler will return true, while the func inside may throw an exception. This may lead to a false positives, but I'd need to check how this would look like in other cases.